demo/summer-2016.html

```diff
@@ -240,14 +239,35 @@
         enableLiveAutocompletion: true
     });
 
+    var HashHandler = require("ace/keyboard/hash_handler").HashHandler; 
+    keyboardHandler = new HashHandler();
+    keyboardHandler.addCommand({
+      name: 'item.',
+      bindKey: {win: 'Enter', mac: 'Enter'},
+      exec: function(editor) {
+        var pos = editor.getCursorPosition()
+        if (editor.getSession().getContext(pos.row) == "list") {
+          var currline = editor.session.getLine(pos.row);
+          if (currline.indexOf("begin") > pos.column) {
+            return false;
+          }
+          editor.insert("\n\t\\item ")
+          return true;
+        }
+        else {
+          return false;
+        }},
+        readOnly: true
```

Comment:
Definitely not, because this command worked perfectly when editor is not read-only=)